src/utils/SharedState.ts

```diff
@@ -35,14 +42,16 @@ export class SharedState<T = any> {
     this.channel.emit(SET_VALUE, key, value, index);
   }
 
-  static subscribe<T>(key: string, channel: Pick<Channel, "emit" | "on" | "off">) {
-    const sharedState = new SharedState<T>(channel);
-    const listeners: ((value: T | undefined) => void)[] = [];
+  static subscribe<T>(key: string, channel: ChannelLike) {
+    const sharedState = instances.get(key) || new SharedState(channel);
 
-    sharedState.channel.on(SET_VALUE, (k: string, v: T | undefined) => {
-      if (k !== key) return;
-      listeners.forEach((listener) => listener(v));
-    });
+    if (!instances.has(key)) {
+      instances.set(key, sharedState);
+      sharedState.channel.on(SET_VALUE, (k: string, v: T | undefined) => {
+        if (k !== key) return;
+        sharedState.listeners.forEach((listener) => listener(v));
+      });
+    }
```

Comment:
It looks like this should use https://github.com/chromaui/addon-visual-tests/pull/138 (refactor-state-management) as the base

Reply:
What makes you think so? SharedState is a separate concept, unrelated to useContext which is applied in #138.

What I did here was avoid creating multiple instances of SharedState each time a component subscribes to it.